code/game/objects/effects/portals.dm

```diff
@@ -47,3 +59,76 @@
 			do_teleport(M, locate(rand(5, world.maxx - 5), rand(5, world.maxy -5), 3), 0)
 		else
 			do_teleport(M, target, precision)
+
+/obj/effect/portal/spawner
+	name = "portal"
+	desc = "Looks like a one-way portal, don't come too close."
+	description_info = "This portal is a spawner portal. You cannot enter it to teleport, but it will periodically spawn things."
+	does_teleport = FALSE
+	has_lifespan = FALSE
+	layer = OBJ_LAYER - 0.01
+	var/list/spawn_things = list() // The list things to spawn
+	var/num_of_spawns			   // How many times we want to spawn them before qdel
+	var/next_spawn
+
+/obj/effect/portal/spawner/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+	next_spawn = world.time + 5 SECONDS
+
+/obj/effect/portal/spawner/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/obj/effect/portal/spawner/process()
+	if(next_spawn < world.time)
+		visible_message(SPAN_WARNING("\The [src] spits something out!"))
+		for(var/thing in spawn_things)
+			var/spawn_path = text2path(thing)
+			var/atom/spawned_thing = new spawn_path(get_turf(src))
+
+			if(istype(spawned_thing, /obj/item/stack))
+				var/obj/item/stack/stack = spawned_thing
+				stack.amount = spawn_things[thing]
+			else
+				for(var/i = 1; i < spawn_things[thing]; i++)
+					new spawn_path(get_turf(src))
```

Comment:
You don't have to do this but I think it would be way cooler if it threw the item instead of just spawning it on the turf.